packages/backend-data/src/factory.ts

```diff
@@ -152,7 +152,10 @@ class DataGenerator implements ConstructContainerEntryGenerator {
       authorizationModes,
       outputStorageStrategy: this.outputStorageStrategy,
       functionNameMap,
-      translationBehavior: { sandboxModeEnabled },
+      translationBehavior: {
+        sandboxModeEnabled,
+        allowDestructiveGraphqlSchemaUpdates: sandboxModeEnabled,
```

Comment:
This should probably just be 
```suggestion
        allowDestructiveGraphqlSchemaUpdates: true,
```
We've had several discussions about this but we've landed on stopping updates like this is the job of CI/CD (eg a check in the amplify build and deploy process) rather than the job of the IaC definition. Reasoning is that if you actually want to make a breaking change, you shouldn't have to update your backend definition to "allow" it and then set it back to "disallowed"